File: app/src/main/java/com/coinninja/coinkeeper/service/client/model/CNTopic.java
package com.coinninja.coinkeeper.service.client.model;

public class CNTopic {
    String id;

    public CNTopic() {
    }

    public CNTopic(String id) {
        this.id = id;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == this) return true;

        if (obj == null || !(obj instanceof CNTopic)) return false;

        return id.equals(((CNTopic) obj).getId());
    }
}

File: app/src/main/java/com/coinninja/coinkeeper/service/client/model/CNTopic.kt
package com.coinninja.coinkeeper.service.client.model

import java.util.*

data class CNTopic(
        var id: String = "",
        var name: String = ""
) {
    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val cnTopic = o as CNTopic?
        return id == cnTopic!!.id
    }

    override fun hashCode(): Int {
        return Objects.hash(id)
    }
}
